codegen/winterfell/src/air/mod.rs

```diff
@@ -1,10 +1,16 @@
 use super::{AirIR, Impl, Scope};
-use air_script_core::{Constant, ConstantType, Expression};
-use ir::IntegrityConstraintDegree;
+use air_script_core::{Constant, ConstantType};
+use ir::{
+    constraints::{ConstantValue, ConstraintDomain, Operation},
+    IntegrityConstraintDegree, NodeIndex, PeriodicColumns,
+};
 
 mod constants;
 use constants::add_constants;
 
+mod ir_types;
```

Comment:
Yeah, I don't like this name either, I just didn't have any better ideas at the time

Reply:
I moved `ElemType`, but otherwise I haven't changed this. Basically, the problem with `codegen_trait` is that we actually have different codegen traits in a couple places (for constants and for periodic columns).  The problem with naming it something codegen-related is that everything in this crate is codegen-related, so I'm not sure that's useful. The idea of this name is that this is where the codegen is defined for the core types of the `ir`, but that's not even fully true. What is true is that it's where there's codegen that's related to the graph/constraints, so maybe I could change it to something about those?

Other thoughts/suggestions would be great

Reply:
> What is true is that it's where there's codegen that's related to the graph/constraints, so maybe I could change it to something about those?

Makes sense. So, maybe just `graph` (seems to be inline with naming of other files)? I don't have a strong opinion though.

Reply:
Yeah, I think there might not be a great answer, but `graph` does seem better.